### ISSUE TITLE
trigger custom code on events?

### ISSUE THREAD

**Issue by mortee** [openmediavault/openmediavault#1389]
Is there a way to execute custom code on certain events, like starting/finish of scheduled jobs, or e.g. an USB Backup run?

The scripts that OMV generates to execute these tasks might check for existence of a script at a pre-defined location, and execute it with passing relevant arguments.

**Comment by votdev** [comment]
No, and i don't think it makes sense because each scheduled job, e.g. USB backup, rsync or virus scan, has a special purpose.

**Comment by mortee** [comment]
My specific real-life scenario is this: I want to do automation when USB Backup has finished. I think that very much makes sense.

**Comment by votdev** [comment]
I fully understand that, but i do not see a way how to implement the management of such "scripts" that are executed pre/post a scheduled job in a way I LIKE it to see it in the UI. I have a vision how the UI and workflows should look like and i do not see that here. I simply DON'T want to see a dumb UI that simply manages shell scripts because this requires knowledge about which userland apps are installed and how shell scripts are build. Something OMV is not designed for.

**Comment by mortee** [comment]
I see your point, fair enough. In any case, there could be a script location which those processes check and execute, and in case the user is knowledgeable enough for it, they could use it. Somewhat like how the OMV env vars are advanced, non-UI-exposed features, still I make very good use of these. That would be trivial to implement, I guess.

For what it's worth, I'd be totally fine with having those events as triggers in the Home Assistant OMV intergartion, and then OMV could fully externalize the whole ordeal. I guess though that adding an event queue to whatever interface these two systems use to commmunicate, would _not_ be trivial...

**Comment by mortee** [comment]
Just for the record, I got around this using a `/etc/rsyslog.d` entry, calling a shell script whenever a line matching my criteria is logged. Still would be nicer if I wouldn't have to resort to scavenging through syslog to catch my events.
